synapse/lib/json.py

```diff
@@ -0,0 +1,191 @@
+import io
+import os
+import mmap
+
+import orjson
+
+import synapse.exc as s_exc
+
+def loads(s):
+    '''
+    Deserialize a JSON string.
+
+    Similar to the standard library json.loads().
+
+    Arguments:
+        s (str | bytes): The JSON data to be deserialized.
+
+    Returns:
+        (object): The deserialized JSON data.
+
+    Raises:
+        synapse.exc.BadJsonText: This exception is raised when there is an error
+            deserializing the provided data.
+    '''
+    try:
+        return orjson.loads(s)
+    except orjson.JSONDecodeError as exc:
+        raise s_exc.BadJsonText(mesg=exc.args[0])
+
+def load(fp):
+    '''
+    Deserialize JSON data from a file.
+
+    Similar to the standard library json.load().
+
+    Arguments:
+        fp (file): The python file pointer to read the data from.
+
+    Returns:
+        (object): The deserialized JSON data.
+
+    Raises:
+        synapse.exc.BadJsonText: This exception is raised when there is an error
+            deserializing the provided data.
+    '''
+    if hasattr(fp, 'mode'):
```

Comment:
Is this a supported trick to determine "real" files from things which just implement the interface?

Reply:
Probably not supported but yeah that was the intent. Fixed now.